lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
> I disagree. Language is important for clarity, especially given that others may need to maintain this in the future (no pun intended, but haha).

I mean, we can change the terms here -- they are not the important part of the code. The concepts are what are important. Promise / Future and often used interchangeably, so changing the terms here is totally fair game, and I agree important, but also easy to do.

> What I think would actually be more appropriate is a separate class which is composed of a future (read has-a vs. is-a)

This is fine with me, and earlier why I suggested a construct with an `Future<T> asFuture();` method or something of the sort on the thing that allowed adding callbacks. It's worth noting though that the API example returns a `Promise` but the implementation in question returns a `Future`... the callback/transform chaining is implementation detail -- if we go with the Future approach, it's important IMHO that the only thing leaked out of the API is `Future`.